controllers/AuthController.js

```diff
@@ -21,77 +21,111 @@ const { CONFIG } = require("./../helpers/config");
-			return UserModel.findOne({ email: value }).then((user) => {
-				if (user) {
-					return Promise.reject("E-mail already in use");
-				}
-			});
-		}),
-	body("password").isLength({ min: 6 }).trim().withMessage("Password must be 6 characters or greater."),
-	body("userType").isLength({ min: 5 }).trim().withMessage("User Type must be 5 characters or greater."),
-	// Sanitize fields.
-	sanitizeBody("name").escape(),
-	sanitizeBody("email").escape(),
-	sanitizeBody("password").escape(),
-	sanitizeBody("userType").escape(),
-	// Process request after validation and sanitization.
-	(req, res) => {
-		try {
-			// Extract the validation errors from a request.
-			const errors = validationResult(req);
-			if (!errors.isEmpty()) {
-				// Display sanitized values/errors messages.
-				return apiResponse.validationErrorWithData(res, "Validation Error.", errors.array());
-			} else {
-				//hash input password
-				bcrypt.hash(req.body.password, 10, function (err, hash) {
-					// generate OTP for confirmation
-					let otp = utility.randomNumber(4);
-					// Create User object with escaped and trimmed data
-					var user = new UserModel(
-						{
-							name: req.body.name,
-							email: req.body.email,
-							password: hash,
-							confirmOTP: otp,
-							userType: req.body.userType,
-						}
-					);
-					// Html email body
-					let html = "<p>Please Confirm your Account.</p><p>OTP: " + otp + "</p>";
-					// Send confirmation email
-					mailer.send(
-						constants.confirmEmails.from,
-						req.body.email,
-						"Confirm Account",
-						html
-					).then(function () {
-						// Save user.
-						user.save(function (err) {
-							if (err) { return apiResponse.ErrorResponse(res, err); }
-							let userData = {
-								_id: user._id,
-								firstName: user.firstName,
-								lastName: user.lastName,
-								email: user.email
-							};
-							return apiResponse.successResponseWithData(res, "Registration Success.", userData);
-						});
-					}).catch(err => {
-						console.log(err);
-						return apiResponse.ErrorResponse(res, err);
-					});
-				});
-			}
-		} catch (err) {
-			//throw error in json response with status 500.
-			return apiResponse.ErrorResponse(res, err);
-		}
-	}];
+  // Validate fields.
+  body("name")
+    .isLength({ min: 1 })
+    .trim()
+    .withMessage("Name must be specified."),
+  body("email")
+    .isLength({ min: 1 })
+    .trim()
+    .withMessage("Email must be specified.")
+    .isEmail()
+    .withMessage("Email must be a valid email address.")
+    .custom((value) => {
+      return UserModel.findOne({ email: value }).then((user) => {
+        if (user) {
+          return Promise.reject("E-mail already in use");
+        }
+      });
+    }),
+  body("password")
+    .isLength({ min: 6 })
+    .trim()
+    .withMessage("Password must be 6 characters or greater."),
+  body("userType")
+    .isLength({ min: 5 })
```

Comment:
Codacy has a fix for the issue: [Expected indentation of 2 tabs but found 4 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7771833)
```suggestion
		.isLength({ min: 5 })
```